File: ios/Classes/Helpers/HeadersEnum.swift
//
//  HeadersEnum.swift
//  backendless_sdk
//
//  Created by Andrii Bodnar on 8/14/19.
//

import Foundation

enum HeadersEnum: String, CaseIterable {
    case userToken = "user-token"
    case loggedIn = "logged-in"
    case sessionTimeout = "session-time-out"
    case appTypeName = "application-type"
    case apiVersion = "api-version"
    case uiState = "uiState"
    
    init?(index: Int) {
        if HeadersEnum.allCases.indices.contains(index) {
            self = HeadersEnum.allCases[index]
        } else {
            return nil
        }
    }
}

File: lib/utils/headers_enum.dart
part of backendless_sdk;

enum HeadersEnum {
  userToken,
  loggedIn,
  sessionTimeOut,
  appType,
  apiVersion,
  uiState
}

extension HeadersEnumExt on HeadersEnum {
  String get header => const {
        HeadersEnum.userToken: 'user-token',
        HeadersEnum.loggedIn: "logged-in",
        HeadersEnum.sessionTimeOut: "session-time-out",
        HeadersEnum.appType: "application-type",
        HeadersEnum.apiVersion: "api-version",
        HeadersEnum.uiState: "uiState",
      }[this]!;
}
